gpii/node_modules/transformer/src/js/Transformer.js

```diff
@@ -25,7 +25,54 @@ fluid.registerNamespace("gpii.transformer");
 
 (function () {
     /**
-     * Helper function for `gpii.transformer.configurationToSettings`. Used to modify the settings handler
+     * Helper function to find out if the capabilitiesTransformations for a settingsHandler block has inputPaths
+     * to handle given settings. If so, return true, otherwise, return false. Setting paths to compare include:
+     * 1. All individual keys in the settings object;
+     * 2. All concatenated settings nested key paths.
+     * In an example of this input settings:
+     * {
+     *     "http://registry.gpii.net/applications/com.microsoft.windows.magnifier": {
+     *         "invertColours": true
+     *     }
+     * }
+     * Setting paths are:
+     * [
+     *     "http://registry.gpii.net/applications/com.microsoft.windows.magnifier",
+     *     "invertColours",
+     *     "http://registry\\.gpii\\.net/applications/com\\.microsoft\\.windows\\.magnifier/invertColours",
+     * ]
+     *
+     * @param {Object} capabilitiesTransformations - The capabilitiesTransformations block for a settingsHandler.
+     * @param {Object} settings - The settings.
+     * Usually from matchMakerOutput.inferredConfiguration.(context).applications.(solutionId).settings.
+     * @return {Boolean} - True if an inputPath is found, otherwise, return false.
+     */
+    gpii.transformer.hasSupportedSettings = function (capabilitiesTransformations, settings) {
+        if (!capabilitiesTransformations || !settings) {
+            return false;
+        }
+        var supportedSettingPaths = fluid.model.transform.collectInputPaths(capabilitiesTransformations);
+        // find concatenated settings nested key paths
+        var actualSettingPaths = gpii.matchMakerFramework.utils.computeLeaves(settings, true);
+
+        // add each key as a path to actualSettingPaths
+        fluid.each(settings, function (value, key) {
+            actualSettingPaths.push(fluid.pathUtil.composeSegments(key));
+            if (fluid.isPlainObject(value, true)) {
+                fluid.each(value, function (subValue, subKey) {
+                    actualSettingPaths.push(fluid.pathUtil.composeSegments(subKey));
+                });
+            }
+        });
+
+        // Find out if one or more actual settings are supported
+        var originalLength = supportedSettingPaths.length;
+        gpii.transformer.removeElement(supportedSettingPaths, actualSettingPaths);
```

Comment:
ES5 primitives have improved sufficiently that it is probably better to express this as a combination of "filter" and "includes" as seen in https://stackoverflow.com/a/19957463

Reply:
OK, ignore this since I see the utility is widely used in the code